packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol

```diff
@@ -9,13 +9,43 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
     using FuelBlockHeaderLiteLib for FuelBlockHeaderLite;
 
     error MessageBlacklisted();
-    error WithdrawalsPaused();
     error MessageRelayFailed();
+    error RateLimitExceeded();
+    error WithdrawalsPaused();
+
+    /// @dev The rate limit setter role
+    bytes32 public constant SET_RATE_LIMITER_ROLE = keccak256("SET_RATE_LIMITER_ROLE");
+
+    /// @notice Duration after which rate limit resets.
+    uint256 public immutable rateLimitDuration;
+
+    /// @notice Amounts already withdrawn this period.
+    uint256 public currentPeriodAmount;
```

Comment:
To allow an upgrade with respect to our current deployment, new state variables have to come after the last state variable, ie. `messageIsBlacklisted`